packages/react/src/components/FluidTimePicker/FluidTimePicker.tsx

```diff
@@ -96,15 +101,26 @@ const FluidTimePicker = React.forwardRef<
     <div className={classNames}>
       <div className={`${prefix}--time-picker--fluid__wrapper`}>
         <div className={`${prefix}--time-picker__input`}>
-          <FluidTextInput ref={ref} {...other} />
+          <FluidTextInput
+            ref={ref}
+            readOnly={readOnly}
+            disabled={disabled}
+            {...other}
+          />
         </div>
         {disabled
           ? React.Children.toArray(children).map((child) => {
               return React.cloneElement(child as React.ReactElement, {
-                disabled,
+                disabled: true,
               });
             })
-          : children}
+          : readOnly
+            ? React.Children.toArray(children).map((child) => {
+                return React.cloneElement(child as React.ReactElement, {
+                  readOnly: true,
+                });
+              })
+            : children}
```

Comment:
it might be worth extracting this to a function now (like how we render `errorText`). what do you think? for example:

```tsx
const childrenWithProps = () => {
  if (disabled) {
    return React.Children.toArray(children).map((child) =>
      React.cloneElement(child as React.ReactElement, {
        disabled: true,
      })
    );
  }
  if (readOnly) {
    return React.Children.toArray(children).map((child) =>
      React.cloneElement(child as React.ReactElement, {
        readOnly: true,
      })
    );
  }
  return children;
};
```